src/providers/postgres/qgspostgresconn.cpp

```diff
@@ -1057,6 +1059,16 @@ QString QgsPostgresConn::quotedValue( const QVariant &value )
   }
 }
 
+QString QgsPostgresConn::quotedJsonValue( const QVariant &value )
+{
+  if ( value.isNull() || !value.isValid() )
+    return QStringLiteral( "null" );
+  if ( value.type() == QVariant::Bool )
+    return value.toBool() ?  QStringLiteral( "true" ) : QStringLiteral( "false" );
+  const auto j { QgsJsonUtils::jsonFromVariant( value ) };
+  return QStringLiteral( "'%1'" ).arg( QString::fromStdString( j.dump() ) );
```

Comment:
Should QgsPostgresConn::quotedValue be used here? Or has escaping quotes already been done by j.dump()?

Reply:
j.dump() returns valid JSON, and quotedValue() would return `NULL` instead of `null` and `TRUE` instead of `true` moreover it format strings, for this reason it's not suitable here but `quotedString()` is.
I've added a test for values that require escaping.